wasm/src/main/scala/ir2wasm/WasmExpressionBuilder.scala

```diff
@@ -1735,4 +1738,30 @@ private class WasmExpressionBuilder private (
 
     IRTypes.AnyType
   }
+
+  private def genClone(t: IRTrees.Clone): IRTypes.Type = {
+    val expr = fctx.addSyntheticLocal(TypeTransformer.transformType(t.expr.tpe)(ctx))
+    fctx.block(Types.WasmRefType(Types.WasmHeapType.ObjectType)) { ourObject =>
+      genTreeAuto(t.expr)
+      instrs += BR_ON_CAST(
+        CastFlags(false, false),
+        ourObject,
+        HeapType(Types.WasmHeapType.Simple.Any),
+        HeapType(Types.WasmHeapType.ObjectType)
+      )
+      instrs += UNREACHABLE
```

Comment:
`BR_ON_CAST` followed by `UNREACHABLE` is the same as `REF_CAST`.

That said, this shouldn't even be necessary, since the argument of a `Clone` node must be statically a subtype of `Cloneable`, and therefore it should already have the Wasm type of `Cloneable`, which is `HeapType(ObjectType)`. (it could be `null`, though)

Reply:
Oh, that's right, because it's "unreachable" 😅

Reply:
https://github.com/tanishiking/scala-wasm/pull/36/commits/6e9b3e6d530257d7319d6cf855df5e258184f5d9